wpilibc/src/main/native/cpp/AnalogInput.cpp

```diff
@@ -48,6 +50,20 @@ AnalogInput::~AnalogInput() {
   m_port = HAL_kInvalidHandle;
 }
 
+AnalogInput::AnalogInput(AnalogInput&& rhs) {
+  m_channel = std::move(rhs.m_channel);
+  std::swap(m_port, rhs.m_port);
```

Comment:
This doesn't work, as m_port is not initialized.

Reply:
It's a shame the HAL handles can't have a default constructor.